src/optimizer/rule/simplification.rs

```diff
@@ -109,6 +119,99 @@ impl Rule for SimplifyFilter {
     }
 }
 
+pub struct LikeRewrite;
+
+impl Rule for LikeRewrite {
+    fn pattern(&self) -> &Pattern {
+        &LIKE_REWRITE_RULE
+    }
+
+    fn apply(&self, node_id: HepNodeId, graph: &mut HepGraph) -> Result<(), OptimizerError> {
+        if let Operator::Filter(mut filter_op) = graph.operator(node_id).clone() {
+            if let ScalarExpression::Binary {
+                op: BinaryOperator::Like,
+                ref mut left_expr,
+                ref mut right_expr,
+                ty,
+            } = filter_op.predicate.clone()
+            {
+                if let ScalarExpression::Constant(value) = right_expr.as_ref() {
+                    if let DataValue::Utf8(value_str) = (**value).clone() {
+                        Self::process_value_str(value_str, left_expr, ty, &mut filter_op);
+                    }
+                }
+            }
+            graph.replace_node(node_id, Operator::Filter(filter_op))
+        }
+        Ok(())
+    }
+}
+
+impl LikeRewrite {
+    fn process_value_str(
+        value_str: Option<String>,
+        left_expr: &mut Box<ScalarExpression>,
+        ty: LogicalType,
+        filter_op: &mut FilterOperator,
+    ) {
+        value_str.map(|value_str| {
+            if value_str.ends_with('%') {
+                let left_bound = value_str.trim_end_matches('%');
+                let right_bound = increment_last_char(left_bound);
+                right_bound.map(|rb| {
+                    filter_op.predicate = Self::create_new_expr(
+                        &mut left_expr.clone(),
+                        ty,
+                        left_bound.to_string(),
+                        rb,
+                    );
+                });
+            }
+        });
+    }
+
+    fn create_new_expr(
+        left_expr: &mut Box<ScalarExpression>,
+        ty: LogicalType,
+        left_bound: String,
+        right_bound: String,
+    ) -> ScalarExpression {
+        let new_expr = ScalarExpression::Binary {
+            op: BinaryOperator::And,
+            left_expr: Box::new(ScalarExpression::Binary {
+                op: BinaryOperator::GtEq,
+                left_expr: left_expr.clone(),
+                right_expr: Box::new(ScalarExpression::Constant(ValueRef::from(DataValue::Utf8(
+                    Some(left_bound),
+                )))),
+                ty,
+            }),
+
+            right_expr: Box::new(ScalarExpression::Binary {
+                op: BinaryOperator::Lt,
+                left_expr: left_expr.clone(),
+                right_expr: Box::new(ScalarExpression::Constant(ValueRef::from(DataValue::Utf8(
+                    Some(right_bound),
+                )))),
+                ty,
+            }),
+            ty,
+        };
+        new_expr
+    }
+}
+
+fn increment_last_char(s: &str) -> Option<String> {
+    let mut chars: Vec<char> = s.chars().collect();
+    for i in (0..chars.len()).rev() {
+        if let Some(next_char) = std::char::from_u32(chars[i] as u32 + 1) {
```

Comment:
I don't think this will continue to work when the last bit is 255
This code doesn't take into account that when carrying out, the last bit needs to be reset to 0! ! ! !